monocle/db/db.py

```diff
@@ -244,156 +246,181 @@ def __init__(
+            "target_branch": {"type": "keyword"},
+            "created_at": {"type": "date", "format": "date_time_no_millis"},
+            "on_created_at": {"type": "date", "format": "date_time_no_millis"},
+            "merged_at": {"type": "date", "format": "date_time_no_millis"},
+            "updated_at": {"type": "date", "format": "date_time_no_millis"},
+            "closed_at": {"type": "date", "format": "date_time_no_millis"},
+            "state": {"type": "keyword"},
+            "duration": {"type": "integer"},
+            "mergeable": {"type": "keyword"},
+            "labels": {"type": "keyword"},
+            "assignees": {
+                "type": "nested",
+                "properties": {
+                    "uid": {"type": "keyword"},
+                    "muid": {"type": "keyword"},
                 },
-            }
+            },
+            "approval": {"type": "keyword"},
+            "draft": {"type": "boolean"},
+            "self_merged": {"type": "boolean"},
         }
-        settings = {"mappings": self.mapping}
+        self.mapping.update(self.new_fields)
+        settings = {"mappings": {"properties": self.mapping}}
         self.ic = self.es.indices
         if create:
-            self.ic.create(index=self.index, ignore=400, body=settings)
-        # The authors_histo is failing on some context with this error when the
-        # time slice is large: Must be less than or equal to: [10000] but was [10001]. ()This limit can be
-        # set by changing the [search.max_buckets] cluster level)
-        # This is an attempt to mitigate the issue
-        cluster_settings = {"transient": {"search.max_buckets": 100000}}
-        self.es.cluster.put_settings(body=cluster_settings)
+            try:
+                self.ic.get(index=self.index)
+            except NotFoundError:
+                self.ic.create(index=self.index, body=settings)
+            # The authors_histo is failing on some context with this error when the
+            # time slice is large: Must be less than or equal to: [10000] but was [10001]. ()This limit can be
+            # set by changing the [search.max_buckets] cluster level)
+            # This is an attempt to mitigate the issue
+            cluster_settings = {"transient": {"search.max_buckets": 100000}}
+            self.es.cluster.put_settings(body=cluster_settings)
+            # Check current index properties and add new filed index properties
+            current_index_properties = (
+                self.ic.get_mapping(index=self.index)
+                .get(self.index, {})
+                .get("mappings", {})
+                .get("properties", {})
+            )
+            for new_field_name, new_field_props in self.new_fields.items():
+                if new_field_name not in current_index_properties:
+                    body = {"properties": {new_field_name: new_field_props}}
+                    self.ic.put_mapping(index=self.index, body=body)
```

Comment:
It seems like we don't need the _schema_version field. Shouldn't this block be performed only when the version is different?

Reply:
Yes let's remove the version field. We don't need it to detect if a new field mapping needs to be added. The logic is already here.